pkg/ebpf/tracer.go

```diff
@@ -796,19 +817,30 @@ func (p *PacketFetcher) ReadPerf() (perf.Record, error) {
 }
 
 func (p *PacketFetcher) LookupAndDeleteMap(c prometheus.Counter) map[int][]*byte {
+	if !p.lookupAndDeleteSupported {
+		return p.legacyLookupAndDeleteMap(c)
+	}
+
 	packetMap := p.objects.PacketRecord
 	iterator := packetMap.Iterate()
 	packets := make(map[int][]*byte, p.cacheMaxSize)
-
 	var id int
+	var ids []int
 	var packet []*byte
+
+	// First, get all ids and ignore content (we need lookup+delete to be atomic)
 	for iterator.Next(&id, &packet) {
-		if err := packetMap.Delete(id); err != nil {
-			log.WithError(err).WithField("packetID ", id).
-				Warnf("couldn't delete  entry")
+		ids = append(ids, id)
+	}
+
+	// Run the atomic Lookup+Delete; if new ids have been inserted in the meantime, they'll be fetched next time
+	for _, id = range ids {
+		if err := packetMap.LookupAndDelete(&id, &packet); err != nil {
+			log.WithError(err).WithField("packetID", id).Warnf("couldn't delete entry")
```

Comment:
Why not using `BatchLookupAndDelete` directly here ?

Reply:
I guess because of:
https://github.com/cilium/ebpf/issues/1078
https://github.com/cilium/ebpf/issues/1080

Reply:
yes there's a follow-up for the batches: https://issues.redhat.com/browse/NETOBSERV-1550